### DISCUSSION TITLE
Create separate playlist for country only and country + international

### DISCUSSION THREAD

**Discussion by Takakeisho** [iptv-org/community#854] [Ideas]
I don't understand why i can't have country only playlist and i have to deal with so many unrelated channels while most of them don't even work. I use multiple playlist on kodi and i have to remove them regularly and half of them are back at the next merge. Why just not creating 2 playlist, one with channels only available in that country for those who use multiple playlists and keeping the actuals one for those who only use one playlist?

**Comment by BellezaEmporium** [comment]
There is a "Grouped by country" section list over on the README, just get the country you want and it's good.

**Reply by Takakeisho** [reply]
Yes, i use those one but by exemple i use the Canadian, i only have around 40 related channels, the rest are just not related at all like Bahrain, Indonesia (berita), China (CGTN & CNA), Germany (DW & NDR), Grece (ERT), Spain (ETB), Europe (Euronews), Taiwan (FTV), Korea (KBS), Mongolia (MNB), Japan (NHK), Russia (RT), Portugal (RTP), Turkey (TRT), Mozambique (TVM). So imagne when i use multiple playlist like this one, The French, Belgian, Swiss, Swedish, Japanese ect. all the repeating channels that i have to remove and most of them are back the next day because someone change the link and i'ts consideered like a new channel.

**Reply by BellezaEmporium** [reply]
> Yes, i use those one but by exemple i use the Canadian, i only have around 40 related channels, the rest are just not related at all like Bahrain, Indonesia (berita), China (CGTN & CNA), Germany (DW & NDR), Grece (ERT), Spain (ETB), Europe (Euronews), Taiwan (FTV), Korea (KBS), Mongolia (MNB), Japan (NHK), Russia (RT), Portugal (RTP), Turkey (TRT), Mozambique (TVM). So imagne when i use multiple playlist like this one, The French, Belgian, Swiss, Swedish, Japanese ect. all the repeating channels that i have to remove and most of them are back the next day because someone change the link and i'ts consideered like a new channel.

I understand it can be frustrating to deal with all the time.
For now it's quite difficult to find a solution for that, because "putting all international channels back to int.m3u" gives it's pack of issues that's painful to deal with.

Have you considered using the API and it's channels.json file ? You may apply some filter with a script that only gives you the channels you want.

**Reply by loucasal** [reply]
> For now it's quite difficult to find a solution for that, because "putting all international channels back to int.m3u" gives it's pack of issues that's painful to deal with.

I take it you are referring to the following (from your own words in iptv-org/iptv#8340):

> The only option that's available on our side is to treat the channels that are streamed internationally as such (by changing their tvg-id as an example, going from .us to .int), but this would cause the EPG not to work, and it would cause a major rework of our playlist creation (and most certainly our database)

How about marking channels as international using a different attribute than their tvg-id? The playlist creation script could then identify channels to be placed in int.m3u based on that attribute, and the EPG would not break. Just a suggestion to help us move away from the current impasse.

**Reply by BellezaEmporium** [reply]
> > For now it's quite difficult to find a solution for that, because "putting all international channels back to int.m3u" gives it's pack of issues that's painful to deal with.
> 
> I take it you are referring to the following (from your own words in iptv-org/iptv#8340):
> 
> > The only option that's available on our side is to treat the channels that are streamed internationally as such (by changing their tvg-id as an example, going from .us to .int), but this would cause the EPG not to work, and it would cause a major rework of our playlist creation (and most certainly our database)
> 
> How about marking channels as international using a different attribute than their tvg-id? The playlist creation script could then identify channels to be placed in int.m3u based on that attribute, and the EPG would not break. Just a suggestion to help us move away from the current impasse.

You're right, this is what I was referring to.

Not quite sure how could we proceed, if we add one more subregion to the concerned streams, or if we just move everything to int.m3u. We still can keep the tvg-id as-is since it won't affect it, but we would need to rework the playlist creation script I think.

Let me check what I can do on my side. @freearhey will most certainly need to have a look on the script that generates the playlists.

**Comment by freearhey** [comment]
Currently, all links in playlists are sorted only based on the `broadcast_area` field from [feeds.csv](https://github.com/iptv-org/database/blob/master/data/feeds.csv), so if some channel is in the wrong playlist, it can be easily corrected through this form: https://github.com/iptv-org/database/issues/new?template=5_feeds_edit.yml

Also now all channels with `r/INT` (International) are excluded from individual country playlists and placed in a separate playlist: 

```
https://iptv-org.github.io/iptv/regions/int.m3u
```
